### PR TITLE
Fix ServicePoint tests for desktop

### PR DESCRIPTION
**Pull request by stephentoub**
Closes https://github.com/dotnet/corefx/issues/18708
Closes https://github.com/dotnet/corefx/issues/18707
cc: @davidsh, @cipop, @Priya91

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test https://github.com/dotnet/corefx/issues/18735 please (Innerloop PortableLinux Release x64 Build and Test)

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop PortableLinux Release x64 Build and Test please (https://github.com/dotnet/corefx/issues/18735)

File: src/System.Net.ServicePoint/tests/ServicePointManagerTest.cs [hunk 1]
```diff
@@ -114,6 +114,7 @@ public static void ReusePort_Roundtrips()
         }
 
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Desktop default SecurityProtocol to Ssl3; explicitly changed to SystemDefault for core.")]
```
Comment (justinvp) [inline src/System.Net.ServicePoint/tests/ServicePointManagerTest.cs]: .NET 4.7's default was changed to `SystemDefault` according to the [release notes](https://github.com/Microsoft/dotnet/blob/master/releases/net47/dotnet47-changes.md#networking) (I haven't verified). Will we want this test to run on .NET Framework when the tests run on 4.7?

Comment (davidsh) [inline src/System.Net.ServicePoint/tests/ServicePointManagerTest.cs]: >Will we want this test to run on .NET Framework when the tests run on 4.7?

Yes.  We will need to figure out how to do runtime detection of .NET Framework  version in these tests to validate against the changed default value.

**Pull request merged**